### ISSUE TITLE
Update dotnet-sdk .deb/.rpm versioning to match dotnet-runtime

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/cli#9404]
Deb and rpm packages for dotnet-runtime-2.1 and dotnet-sdk-2.1 are using different patterns for versioning. Per discussion with @rakeshsinghranchi and @leecow, let's update dotnet-sdk to follow what dotnet-runtime is doing.

For comparison, this is what the packages currently look like.

**Deb**
```
$ dpkg --info ./dotnet-runtime-2.1.1-rtm-26530-02-x64.deb
Version: 2.1.1~rtm-26530-02-1
 
$ dpkg --info ./dotnet-sdk-2.1.301-preview-008887-x64.deb
Version: 2.1.301-preview-008887-1
```

**Rpm**
```
$ rpm -qpiR ./dotnet-runtime-2.1.1-rtm-26530-02-x64.rpm
Version     : 2.1.1
Release     : 0.1.rtm_26530_02
 
$ rpm -qpiR ./dotnet-sdk-2.1.301-preview-008887-x64.rpm
Version     : 2.1.301_preview_008887
Release     : 1
```

cref https://github.com/dotnet/core-setup/issues/3850

**Comment by livarcocc** [comment]
This is different than what I thought. I thought this was about removing the -upgrade dependency for the core-setup assets.

**Comment by natemcmaster** [comment]
Yeah, that's separate. This can wait until 2.1.400 or later. It's not urgent.

**Comment by natemcmaster** [comment]
cc @akoeplinger heads up. When this is resolved, we'll need a slight change to the update I made in https://github.com/travis-ci/travis-build/pull/1397. And by slight, I mean truly slight. We'll need to replace a `-` with a `~`.

**Comment by leecow** [comment]
I would expect this to be fixed in the 2.1.3xx train and forward.
